### PR TITLE
[incubator-kie-issues#1220] Quarkus example to showcase PIM functionality

### PR DESCRIPTION
**Pull request by martinweiler**
Closes https://github.com/apache/incubator-kie-issues/issues/1220

### CODE REVIEW

File: kogito-quarkus-examples/process-instance-migration-quarkus/README.md [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+- Environment variable GRAALVM_HOME set accordingly
+- GraalVM native image needs as well native-image extension: https://www.graalvm.org/reference-manual/native-image/
+- Note that GraalVM native image compilation typically requires other packages (glibc-devel, zlib-devel and gcc) to be installed too, please refer to GraalVM installation documentation for more details.
+
+### Infrastructure Services
+
+This quickstart provides a docker compose template that starts all the required services. This setup ensures that all services are connected with a default configuration.
+
+- PostgreSQL: 5432
+- Data Index: 8180
```
Comment (pefernan) [inline kogito-quarkus-examples/process-instance-migration-quarkus/README.md]: @martinweiler shouldn't we be using the compact setup?

File: kogito-quarkus-examples/process-instance-migration-quarkus/docker-compose/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+      /entrypoint.sh
+      "
+    environment:
+      PGADMIN_DEFAULT_EMAIL: user@kogito.org
+      PGADMIN_DEFAULT_PASSWORD: pass
+      PGADMIN_CONFIG_SERVER_MODE: 'False'
+      PGADMIN_CONFIG_MASTER_PASSWORD_REQUIRED: 'False'
+      GUNICORN_ACCESS_LOGFILE: '/dev/null'
+
+  data-index:
```
Comment (pefernan) [inline kogito-quarkus-examples/process-instance-migration-quarkus/docker-compose/docker-compose.yml]: No needed with compact setup

File: kogito-quarkus-examples/process-instance-migration-quarkus/pom.xml [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+    </dependency>
+    <dependency>
+      <groupId>org.kie</groupId>
+      <artifactId>kie-addons-quarkus-persistence-jdbc</artifactId>
+    </dependency>
+
+    <!-- Data Index Persistence -->
+    <dependency>
+      <groupId>org.kie</groupId>
+      <artifactId>kogito-addons-quarkus-data-index-persistence-postgresql</artifactId>
```
Comment (pefernan) [inline kogito-quarkus-examples/process-instance-migration-quarkus/pom.xml:136]: Change this to the data-index component
```suggestion
      <artifactId>kogito-addons-quarkus-data-index-postgresql</artifactId>
```

File: kogito-quarkus-examples/process-instance-migration-quarkus/docker-compose/docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    environment:
+      QUARKUS_HTTP_CORS_ORIGINS: "/.*/"
+      QUARKUS_DATASOURCE_JDBC_URL: "jdbc:postgresql://postgres:5432/kogito"
+      QUARKUS_DATASOURCE_REACTIVE_URL: "postgresql://postgres:5432/kogito"
+      QUARKUS_DATASOURCE_USERNAME: kogito-user
+      QUARKUS_DATASOURCE_PASSWORD: kogito-pass
+      QUARKUS_DATASOURCE_DB_KIND: postgresql
+      KOGITO_JOBS_SERVICE_URL: http://${DOCKER_GATEWAY_HOST}:8080
+      KOGITO_SERVICE_URL: http://${DOCKER_GATEWAY_HOST}:8080
+      KOGITO_DATAINDEX_HTTP_URL: http://${DOCKER_GATEWAY_HOST}:8180
```
Comment (pefernan) [inline kogito-quarkus-examples/process-instance-migration-quarkus/docker-compose/docker-compose.yml]:
```suggestion
      KOGITO_DATAINDEX_HTTP_URL: http://${DOCKER_GATEWAY_HOST}:8080
```

File: kogito-quarkus-examples/process-instance-migration-quarkus/src/main/resources/application.properties [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+quarkus.http.test-port=0
+
+# Kogito-service
+kogito.service.url=http://localhost:8080
+
+#Job-service
+kogito.jobs-service.url=http://localhost:8080
+
+# to be reachable from the container running job-service
+kogito.dataindex.http.url=http://localhost:8180
```
Comment (pefernan) [inline kogito-quarkus-examples/process-instance-migration-quarkus/src/main/resources/application.properties]:
```suggestion
kogito.dataindex.http.url=http://localhost:8080
```

File: kogito-quarkus-examples/process-instance-migration-quarkus/src/main/resources/META-INF/resources/index.html [hunk 6]
```diff
@@ -0,0 +1,150 @@
+<!--
```
Comment (pefernan) [inline kogito-quarkus-examples/process-instance-migration-quarkus/src/main/resources/META-INF/resources/index.html]: Do we need this page?

File: kogito-quarkus-examples/process-instance-migration-quarkus/docker-compose/sql/init.sql [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+CREATE DATABASE kogito
+    WITH
+    OWNER = "kogito-user"
+    ENCODING = 'UTF8'
+    LC_COLLATE = 'en_US.utf8'
+    LC_CTYPE = 'en_US.utf8'
+    TABLESPACE = pg_default
+    CONNECTION LIMIT = -1;
+
+CREATE DATABASE keycloak
```
Comment (pefernan) [inline kogito-quarkus-examples/process-instance-migration-quarkus/docker-compose/sql/init.sql]: Not needed if we don't have consoles

File: kogito-quarkus-examples/process-instance-migration-quarkus/docker-compose/sql/init.sql [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+    LC_COLLATE = 'en_US.utf8'
+    LC_CTYPE = 'en_US.utf8'
+    TABLESPACE = pg_default
+    CONNECTION LIMIT = -1;
+
+GRANT ALL PRIVILEGES ON DATABASE postgres TO "kogito-user";
+GRANT ALL PRIVILEGES ON DATABASE kogito TO "kogito-user";
+GRANT ALL PRIVILEGES ON DATABASE kogito TO postgres;
+
+GRANT ALL PRIVILEGES ON DATABASE keycloak TO "kogito-user";
```
Comment (pefernan) [inline kogito-quarkus-examples/process-instance-migration-quarkus/docker-compose/sql/init.sql]: same

File: kogito-quarkus-examples/process-instance-migration-quarkus/src/main/resources/META-INF/processSVG/addedtask-svg.svg
Comment (pefernan) [inline kogito-quarkus-examples/process-instance-migration-quarkus/src/main/resources/META-INF/processSVG/addedtask-svg.svg:1]: We may not need this if we don't have consoles

File: kogito-quarkus-examples/process-instance-migration-quarkus/docker-compose/README.md [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+
+Eg:
+```shell
+sh startServices.sh example
+```
+
+Once the services are started (depending on the profile), the following ports will be assigned on your local machine:
+- Postgresql: 5432
+- PgAdmin: 8055
+- Kogito Data Index: 8180
```
Comment (pefernan) [inline kogito-quarkus-examples/process-instance-migration-quarkus/docker-compose/README.md]: Not needed

File: kogito-quarkus-examples/process-instance-migration-quarkus/docker-compose/docker-compose.yml [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+      - "${DOCKER_GATEWAY_HOST}:host-gateway"
+
+  kogito-process-instance-migration-service:
+    container_name: kogito-process-instance-migration-service
+    image: dev.local/${USER}/kogito-process-instance-migration-service:1.0-SNAPSHOT
+    profiles: ["example", "full"]
+    ports:
+      - "8080:8080"
+    depends_on:
+      data-index:
```
Comment (pefernan) [inline kogito-quarkus-examples/process-instance-migration-quarkus/docker-compose/docker-compose.yml]: not needed

File: kogito-quarkus-examples/process-instance-migration-quarkus/pom.xml [hunk 11]
```diff
@@ -0,0 +1,214 @@
...
+    <dependency>
+      <groupId>io.quarkus</groupId>
+      <artifactId>quarkus-agroal</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>org.kie</groupId>
+      <artifactId>kie-addons-quarkus-persistence-jdbc</artifactId>
+    </dependency>
+
+    <!-- Data Index Persistence -->
```
Comment (pefernan) [inline kogito-quarkus-examples/process-instance-migration-quarkus/pom.xml:133]:
```suggestion
    <!-- Data Index -->
```

**Review by pefernan — CHANGES_REQUESTED** [review]
@martinweiler great example!

Just requested a few changes related to some missing bits of the compact setup, basically using the data-index addon in the app (and remove it from the docker-compose). A part from that everything works as expected

**Comment by martinweiler** [comment]
@pefernan thanks for your review. I have updated the example now to be aligned with the compact architecture example. Could you please take another look? Thanks!

**Review by elguardian — APPROVED** [review]
good work.

**Review by pefernan — APPROVED** [review]

**Review by porcelli — APPROVED** [review]

**Pull request merged**